src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/sync/ContainersSync.groovy

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.morpheusdata.nutanix.prismelement.plugin.sync
+
+import com.morpheusdata.core.MorpheusContext
+import com.morpheusdata.core.data.DataQuery
+import com.morpheusdata.core.util.HttpApiClient
+import com.morpheusdata.core.util.SyncTask
+import com.morpheusdata.model.Cloud
+import com.morpheusdata.model.Datastore
+import com.morpheusdata.model.projection.DatastoreIdentity
+import com.morpheusdata.nutanix.prismelement.plugin.NutanixPrismElementPlugin
+import com.morpheusdata.nutanix.prismelement.plugin.utils.NutanixPrismElementApiService
+import groovy.util.logging.Slf4j
+
+/**
+ * A class to sync Nutanix Storage Containers to their Morpheus Datastore equivalent
+ */
+@Slf4j
+class ContainersSync {
+	private final MorpheusContext context
+	private final HttpApiClient client
+	private final Cloud cloud
+
+	ContainersSync(MorpheusContext context, Cloud cloud, HttpApiClient client) {
+		this.context = context
+		this.cloud = cloud
+		this.client = client
+	}
+
+	def execute() {
+		log.info("Executing container sync for cloud ${cloud.name}")
+		try {
+			def authConfig = NutanixPrismElementPlugin.getAuthConfig(context, cloud)
+			def listResults = NutanixPrismElementApiService.listContainers(client, authConfig)
+			if (listResults.success == true) {
+				def objList = listResults?.containers?.findAll {
+					it.name?.toLowerCase() != 'nutanixmanagementshare'
+				}
+
+				def existingItems = context.async.cloud.datastore.listIdentityProjections(
+					new DataQuery()
+						.withFilter('refType', 'ComputeZone')
+						.withFilter('refId', cloud.id)
+						.withFilter('type', 'generic')
+				)
+
+				SyncTask<DatastoreIdentity, Map, Datastore> syncTask = new SyncTask<>(existingItems, objList)
+				syncTask.addMatchFunction { DatastoreIdentity datastoreIdentity, Map cloudItem ->
+					datastoreIdentity?.externalId == cloudItem?.uuid
+				}.onAdd { List<Map> cloudItems ->
+					cloudItems?.each { cloudItem ->
+						log.debug("adding boot policy: ${cloudItem}")
+						context.services.cloud.datastore.create(new Datastore(
+							owner: cloud.owner,
+							code: "nutanix.acropolis.datastore.${cloud.id}.${cloudItem.id}",
+							cloud: cloud,
+							category: "nutanix.acropolis.datastore.${cloud.id}",
+							name: cloudItem.name,
+							internalId: cloudItem.id,
+							externalId: cloudItem.uuid,
+							refType: 'ComputeZone',
+							refId: cloud.id,
+							storageSize: Long.valueOf(cloudItem.maxStorage ?: 0),
+							freeSpace: Long.valueOf(cloudItem.freeStorage ?: 0),
+							active: cloud.defaultDatastoreSyncActive
+						))
+					}
+				}.onDelete { List<Datastore> morpheusItems ->
+					morpheusItems?.each { morpheusItem ->
+						log.debug("removing datastore ${morpheusItem}")
```

Comment:
Use `context.services.cloud.datastore.bulkRemove(morpheusItems)` method here to reduce overhead on API calls.